addons/sale_layout/migrations/8.0.1.0/post-migration.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    This module copyright (C) 2015 Therp BV (<http://therp.nl>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp import SUPERUSER_ID, api
+from openerp.openupgrade import openupgrade
+
+
+def migrate_sale_order_lines(env):
+    env.cr.execute(
+        "select id, layout_type from sale_order_line "
+        "where layout_type not in ('article', 'text')")
+    line_layouts = dict(env.cr.fetchall())
+
+    for sale_order in env['sale.order'].search(
+            [('order_line', 'in', line_layouts.keys())]):
+        current_section = None
+        sequence = 1
+        for line in sale_order.order_line:
+            layout_type = line_layouts.get(line.id)
+            if layout_type == 'title':
+                current_section = env['sale_layout.category'].create({
+                    'name': line.name,
+                    'subtotal': False,
+                    'separator': False,
+                    'pagebreak': False,
+                    'sequence': sequence,
+                })
+                sequence += 1
+                if not line.product_id and line.state in ('draft', 'cancel'):
```

Comment:
in 6.1, you would use a line to only say "I'm a page break" etc. So if they don't do anything useful (checked by product_id), it's better to remove them because they'd mess up the layout.